app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php

```diff
@@ -40,8 +40,9 @@ public function getCategory()
 
     public function getCategoryId()
     {
-        if ($this->getCategory()) {
-            return $this->getCategory()->getId();
+        $category = $this->getCategory();
+        if (is_object($category) && $category->getId()) {
```

Comment:
```suggestion
        if (is_object($category)) {
```

@elidrissidev what about this?

Reply:
Actually now that I think about it, what was reason of changing the `if ($this->getCategory())` condition? The only PHPStan error in this file was about the `Right side of && is always false` which you already fixed below..